src/transformState.js

```diff
@@ -5,7 +5,24 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const arr of action.keysToRemove) {
+          delete state[arr];
+        }
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        };
+    }
```

Comment:
It's a good practice to add default case for error handling to `switch/case` statement